### PR TITLE
ScalaJS Support

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
This is a pure library so should be fairly straightforward.

### CODE REVIEW

**Comment by tpolecat** [comment]
Oh cool, thanks. This makes sense.

File: build.sbt [hunk 1]
```diff
@@ -4,49 +4,75 @@ lazy val `scala-2.12`     = "2.12.12"
...
-  val old = (Compile / doc / sources).value
-  if (scalaVersion.value.startsWith("3."))
-    Seq()
-  else
-    old
-}
-
-enablePlugins(AutomateHeaderPlugin)
+  .jsSettings(
+    scalaJSLinkerConfig ~= (_.withModuleKind(ModuleKind.CommonJSModule))
```
Comment (ChristopherDavenport) [inline build.sbt:75]: Not 100% on this setting

Comment (armanbilge) [reply build.sbt:75]: This is fine! It enables `munit` to import filesystem modules in Node.js. In fact you could make it `Test / scalaJSLinkerConfig`.

**Comment by SethTisue** [comment]
"First-time contributors need a maintainer to approve running workflows"... dunno Rob, do you trust Chris?

**Comment by armanbilge** [comment]
Would be great to see this go through! I'm just about to PR skunk and this is all that it's missing :)

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:75]: Actually if you update to the latest munit this shouldn't be needed anymore.

**Comment by tpolecat** [comment]
subsumed by #7

**Pull request closed without merging**